Tools/CISupport/ews-build/events.py

```diff
@@ -89,25 +95,41 @@ def __init__(self, master_hostname, type_prefix='', name='Events'):
         self.type_prefix = type_prefix
         self.master_hostname = master_hostname
 
-    def sendData(self, data):
+    def sendDataToEWS(self, data):
         if os.getenv('EWS_API_KEY', None):
             data['EWS_API_KEY'] = os.getenv('EWS_API_KEY')
         agent = Agent(reactor)
         body = JSONProducer(data)
 
         agent.request(b'POST', self.EVENT_SERVER_ENDPOINT, Headers({'Content-Type': ['application/json']}), body)
 
+    def sendDataToGitHub(self, repository, sha, data):
+        username, access_token = GitHub.credentials()
+
+        data['description'] = data.get('description', '')
+        if len(data['description']) > self.MAX_GITHUB_DESCRIPTION:
+            data['description'] = '{}...'.format(data['description'][:self.MAX_GITHUB_DESCRIPTION - 3])
+
+        agent = Agent(reactor)
+        body = JSONProducer(data)
+        d = agent.request(b'POST', '{}/statuses/{}'.format(GitHub.api_url(repository), sha).encode('utf-8'), Headers({
+            'Authorization': ['Basic {}'.format(b64encode('{}:{}'.format(username, access_token).encode('utf-8')).decode('utf-8'))],
+            'User-Agent': ['python-twisted/{}'.format(twisted.__version__)],
+            'Accept': ['application/vnd.github.v3+json'],
+            'Content-Type': ['application/json'],
+        }), body)
+
     def getBuilderName(self, build):
         if not (build and 'properties' in build):
             return ''
 
         return build.get('properties').get('buildername')[0]
 
-    def getPatchID(self, build):
-        if not (build and 'properties' in build and 'patch_id' in build['properties']):
+    def getProperty(self, build, property_):
```

Comment:
is this method name intentional? (same as buildbot method name)

Reply:
same name might cause confusion or might be misleading. We can call it something like extractProperty. I guess then we can also rename `property_` to `property`.